src/ResourceManager/Profile/Commands.Profile.Test/TypeConversionTests.cs

```diff
@@ -33,7 +33,7 @@ public TypeConversionTests(ITestOutputHelper output)
         [Trait(Category.AcceptanceType, Category.CheckIn)]
         public void CanConvertNullEnvironments()
         {
-            Assert.Null((PSAzureEnvironment)null);
+            Assert.Null(null);
```

Comment:
Hmmm, so the point here was to test application of the explicit type converter for these types.  It's possible the compiler might optimize these out, but likely only when the explicit conversion is doing what it is supposed to.

Reply:
this would apply to the other applications of this

Reply:
Boxing here would make no difference if you are doing `(Thing)null == null` which is essentially what this is doing, right? You can't actually cast `null`. It just gets boxed. https://dotnetliberty.com/index.php/2015/10/11/cast-to-object-before-comparing-to-null/ AFAIK, `null` is of type Object. Every reference type derives of Object. So, even though you are casting to some reference type, it'll get compared as boxed Object when doing the null check. I could be wrong on that, though.

As an aside, obviously `Assert.Null(null)` will always be true, so optimizing this just looks goofy.